crates/bevy_camera/src/camera.rs

```diff
@@ -610,28 +610,27 @@ impl Camera {
         let target_rect = self
             .logical_viewport_rect()
             .ok_or(ViewportConversionError::NoViewportSize)?;
-        let mut rect_relative = (viewport_position - target_rect.min) / target_rect.size();
+        let rect_relative = (viewport_position - target_rect.min) / target_rect.size();
+        let mut ndc_xy = rect_relative * 2. - Vec2::ONE;
         // Flip the Y co-ordinate origin from the top to the bottom.
-        rect_relative.y = 1.0 - rect_relative.y;
+        ndc_xy.y = -ndc_xy.y;
 
-        let ndc_point_near = (rect_relative * 2. - Vec2::ONE).extend(1.0).into();
+        let ndc_point_near = ndc_xy.extend(1.0).into();
+        // Using EPSILON because an ndc with Z = 0 returns NaNs.
+        let ndc_point_far = ndc_xy.extend(f32::EPSILON).into();
+        let view_from_clip = self.computed.clip_from_view.inverse();
+        let world_from_view = camera_transform.affine();
         // We multiply the point by `view_from_clip` and then `world_from_view` in sequence to avoid the precision loss
         // (and performance penalty) incurred by pre-composing an affine transform with a projective transform.
         // Additionally, we avoid adding and subtracting translation to the direction component to maintain precision.
```

Comment:
The comments is not valid anymore

Reply:
Yeah, sorry, I completely misread the comment and the code. Makes sense to me now. For some reason I didn't see all the `world_from_view` so I thought it was outdated.